src/components/WorkBuddies/MyWorkBuddy.tsx

```diff
@@ -0,0 +1,118 @@
+import { Fragment, useContext, useEffect, useState } from "react";
+import { BuddiesContext } from "../BuddiesContext";
+import classNames from "classnames";
+import "./work-buddies.css";
+import { useReducer } from "react";
+import { Buddy } from "./BuddyInterface";
+import { memo } from "react";
+
+interface propsInterface {
+  myBuddy: Buddy;
+}
+
+const reducer = (
+  index: number,
+  action: { type: string; contextData?: Buddy[]; myBuddyIndex?: number }
+) => {
+  switch (action.type) {
+    case "next":
+      if (action.contextData) {
+        if (index + 1 < action.contextData.length) return index + 1;
+        else return 0;
+      } else return 0;
+
+    case "prev":
+      if (action.contextData) {
+        if (index - 1 > -1) return index - 1;
+        else return action.contextData.length - 1;
+      } else return 0;
+
+    case "my-buddy":
+      if (action.myBuddyIndex) {
+        return action.myBuddyIndex;
+      } else return 0;
+
+    default:
+      return 1;
+  }
+};
+
+const MyWorkBuddy = memo(({ myBuddy }: propsInterface) => {
+  const { contextData } = useContext(BuddiesContext);
+
+  //useReducer
+  const [currentBuddyIndex, dispatch] = useReducer(reducer, 0);
+  const [buddy, setBuddy] = useState<Buddy>();
```

Comment:
If I remove this line and the bellow useEffect, would anything change in the 
application?
![image](https://github.com/TheaAleksovDev/ReactConceptsPractice/assets/26555796/abeab878-969a-4e18-a03e-afea2840887f)

The reason I'm asking is because I don't see the `buddy` variable used anywhere.
It's almost as if you are setting a state and not using it.

Reply:
It was just for practice of using useMemo(), but after reading other suggestions I changed it to name:string instead of myBuddy, and now I'm using it